rpc.go

```diff
@@ -113,6 +114,28 @@ type libvirtError struct {
 	Level    uint32
 }
 
+func (e libvirtError) Error() string {
+	return fmt.Sprintf("%s, code: %d, domain id: %d", e.Message, e.Code, e.DomainID)
+}
+
+func checkError(err error, expectedError ErrorNumber) bool {
+	e, ok := err.(libvirtError)
+	if ok {
+		return e.Code == uint32(expectedError)
+	}
+	return false
+}
+
+// ErrIsNotFound detects libvirt's ERR_NO_DOMAIN.
+func ErrIsNotFound(err error) bool {
```

Comment:
Nit: I'd drop the "Err" prefix, and the typical style would be "IsNotExist", as is done in `os` and the like: https://golang.org/pkg/os/#IsNotExist

But I'll leave it up to @trapgate if he'd prefer this or not.

Reply:
Good suggestion, thanks!